taf/developer_tool.py

```diff
@@ -1272,6 +1282,61 @@ def setup_test_yubikey(key_path=None):
     print("Pin: {}\n".format(pin))
 
 
+def update_and_sign_targets(
+    repo_path: str,
+    library_dir: str,
+    target_types: list,
+    keystore: str,
+    roles_key_infos: str,
+    no_commit: bool,
+    scheme: str,
+):
+    """
+    <Purpose>
+        Save the top commit of specified target repositories to the corresponding target files and sign
+    <Arguments>
+        repo_path:
+        Authentication repository's location
+        library_dir:
+        Directory where target repositories and, optionally, authentication repository are locate
+        targets:
+        Types of target repositories whose corresponding target files should be updated and signed
+        keystore:
+        Location of the keystore files
+        roles_key_infos:
+        A dictionary whose keys are role names, while values contain information about the keys
+        no_commit:
+        Indicates that the changes should bot get committed automatically
+        scheme:
+        A signature scheme used for signing
+
+    """
+    auth_path = Path(repo_path).resolve()
+    auth_repo = AuthenticationRepository(path=auth_path)
+    if library_dir is None:
+        library_dir = auth_path.parent.parent
+    repositoriesdb.load_repositories(auth_repo)
+    signed_targets = []
+    for target_type in target_types:
+        try:
+            target_name = repositoriesdb.get_repositories_paths_by_custom_data(
+                auth_repo, type=target_type
+            )[0]
+        except Exception:
+            print(
+                f"Skipping {target_type}. Type listed in repositories.json as a target repository"
+            )
+            continue
```

Comment:
Should we raise an error when we specify `target_type` and it's not found, or do we want to continue and sign other targets? If we're skipping `target_types` which don't exist in `repositories.json`, would it be simpler to just sign the types which exist in repositories.json (and not need to specify types in the cli)?

Reply:
> Maybe we should check if they all exist before updating anything? In this case, if you make a typo, you'd have to run the command again

Okay, makes sense

> We already have a command which updates everything listed in repositories.json, but it proved to be quite problematic. 

Which command is that?

Reply:
uodate-repos-from-repositories-json